SDWebImage/Core/SDImageIOAnimatedCoder.m

```diff
@@ -108,6 +108,32 @@ + (NSUInteger)defaultLoopCount {
 
 #pragma mark - Utils
 
++ (BOOL)canDecodeFromFormat:(SDImageFormat)format {
+    CFStringRef imageUTType = [NSData sd_UTTypeFromImageFormat:format];
+    NSArray *imageUTTypes = (__bridge_transfer NSArray *)CGImageSourceCopyTypeIdentifiers();
+    if ([imageUTTypes containsObject:(__bridge NSString *)(imageUTType)]) {
+        // Can decode from target format
+        return YES;
+    }
+    return NO;
```

Comment:
How about change `NSArray *imageUTTypes` to  `static  NSSet *imageUTTypes`, In this way, other coders can directly call the contain method, and each coder does not need to declare its own internal method, like:
```objc
+ (BOOL)canDecodeFromWebPFormat;
+ (BOOL)canDecodeFromHEICFormat;
+ (BOOL)canDecodeFromHEIFFormat;
```

Reply:
It's OK with this optimization to use RAM to trade off the CPU. But the access to Set seems not performent than use static once method.

Reply:
We can still keep these methods. Does this cause something issue ?

Reply:
I mean, It's just that you don't need to declare it in an internal file, call the unified method directly.  Thus, If you add a new coder, you don't need add the `canDecodeFromNewFormat` && `canEncodeToNewFormat`  method in internal file.